Library/Homebrew/formula.rb

```diff
@@ -3102,6 +3114,50 @@ def license(args = nil)
       end
     end
 
+    # @!attribute [w] allow_network_access_for_build
+    # Allows network access within the sandbox used during formula build.
+    def allow_network_access_for_build
+      @network_access_allowed[:build] = true
+    end
+
+    # @!attribute [w] deny_network_access_for_build
+    # Denies network access within the sandbox used during formula build.
+    def deny_network_access_for_build
+      @network_access_allowed[:build] = false
+    end
+
+    # @!attribute [w] allow_network_access_for_postinstall
+    # Allows network access within the sandbox used during formula postinstall.
+    def allow_network_access_for_postinstall
+      @network_access_allowed[:postinstall] = true
+    end
+
+    # @!attribute [w] deny_network_access_for_postinstall
+    # Denies network access within the sandbox used during formula postinstall.
+    def deny_network_access_for_postinstall
+      @network_access_allowed[:postinstall] = false
+    end
+
+    # @!attribute [w] allow_network_access_for_test
+    # Allows network access within the sandbox used during formula test.
+    def allow_network_access_for_test
+      @network_access_allowed[:test] = true
+    end
+
+    # @!attribute [w] allow_network_access_for_test
+    # Denies network access within the sandbox used during formula test.
+    def deny_network_access_for_test
+      @network_access_allowed[:test] = false
+    end
```

Comment:
Not a super strong opinion, but I preferred what you did originally here, except for the name. i.e. you can call `deny_network_access` if you want to deny it everywhere, or `deny_network_access [:build, :postinstall]` if you want to allow it during `brew test`.